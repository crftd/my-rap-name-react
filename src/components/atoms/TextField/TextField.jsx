import React from 'react';
import TextField from '@material-ui/core/TextField';

export default ({ field, form, ...other }) => (
  <TextField
    style={{ width: '100%' }}
    label="Name"
    helperText={form.errors[field.name]}
    error={Boolean(form.errors[field.name])}
    {...field}
    {...other}
  />
);
