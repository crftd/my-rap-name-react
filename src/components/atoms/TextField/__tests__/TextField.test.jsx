import React from 'react';
import renderer from 'react-test-renderer';

import TextField from '../TextField.jsx';

describe('[Snapshot] TextField', () => {
  it('should render', () => {
    // Arrange
    const expectedForm = {
      errors: {},
    };
    const expectedField = {
      onChange: () => {},
      onBlur: () => {},
      name: '',
      value: '',
    };

    // Act
    const actualTree = renderer.create(<TextField field={expectedField} form={expectedForm} />).toJSON();

    // Assert
    expect(actualTree).toMatchSnapshot();
  });
});
