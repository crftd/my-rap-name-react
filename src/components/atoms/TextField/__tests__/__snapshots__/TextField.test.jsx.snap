// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] TextField should render 1`] = `
<div
  className="MuiFormControl-root-1"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
  >
    Name
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
